spec/rubocop/cop/lint/unused_block_argument_spec.rb

```diff
@@ -308,6 +308,190 @@ def other(a)
+              [ { foo: 1, bar: 2 } ].each do |foo:, bar:|
+                                                    ^^^ #{bar_message}
+                                              ^^^ #{foo_message}
+                puts :something
+              end
+            RUBY
+          end
+        end
+
+        context 'when there is trailing double splat' do
+          it 'does not register an offense' do
+            expect_no_offenses(<<-RUBY.strip_indent)
+              [ { foo: 1, bar: 2 } ].each do |foo:, **|
+                puts foo
+              end
+            RUBY
+          end
+        end
+      end
+
+      context 'when a block takes single argument' do
+        context 'and the argument is unused' do
+          it 'registers an offense and suggests omitting that' do
+            message = 'Unused block argument - `a`. ' \
+                        "You can omit the argument if you don't care about it."
+
+            expect_offense(<<-RUBY.strip_indent)
+               [{a: 1}].each do |a:|
+                                 ^ #{message}
+                puts :something
+              end
+            RUBY
+          end
+        end
+
+        context 'and the method call is `define_method`' do
+          it 'registers an offense' do
+            message = 'Unused block argument - `bar`. ' \
+                        "If it's necessary, use `_` or `_bar` as an argument " \
+                        "name to indicate that it won't be used."
+
+            expect_offense(<<-RUBY.strip_indent)
+              define_method(:foo) do |bar:|
+                                      ^^^ #{message}
+                puts 'baz'
+              end
+            RUBY
+          end
+        end
+      end
+
+      context 'when a method argument is not used' do
+        it 'does not care' do
+          expect_no_offenses(<<-RUBY.strip_indent)
+            def some_method(foo:)
+            end
+          RUBY
+        end
+      end
+
+      context 'in a method calling `binding` without arguments' do
+        it 'accepts all arguments' do
+          expect_no_offenses(<<-RUBY.strip_indent)
+            test do |key:, value:|
+              puts something(binding)
+            end
+          RUBY
+        end
+
+        context 'inside a method definition' do
+          it 'registers offenses' do
+            (key_message, value_message) = %w[key value].map do |arg|
+              "Unused block argument - `#{arg}`. You can omit all the " \
+                "arguments if you don't care about them."
+            end
+
+            expect_offense(<<-RUBY.strip_indent)
+              test do |key:, value:|
+                             ^^^^^ #{value_message}
+                       ^^^ #{key_message}
+                def other(a)
+                  puts something(binding)
+                end
+              end
+            RUBY
+          end
+        end
+      end
+
+      context 'in a method calling `binding` with arguments' do
```

Comment:
And so is this one IMO. The context is not really "in a method calling `binding`".

Reply:
same as above.